locator/tablets.cc

```diff
@@ -717,15 +731,53 @@ class tablet_effective_replication_map : public effective_replication_map {
     }
 
 public:
+
     tablet_effective_replication_map(table_id table,
                                      replication_strategy_ptr rs,
                                      token_metadata_ptr tmptr,
-                                     size_t replication_factor)
+                                     size_t replication_factor,
+                                     absl::flat_hash_map<sstring, replication_factor_index_t>& datacenter_map,
+                                     utils::chunked_vector<replication_factor_list>& rf_map)
```

Comment:
It may be surprising that things passed by ref are moved-from. Better to capture by && and require explicit std::move() in the caller.